homeassistant/components/tuya/config_flow.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python3
+"""Config flow for Tuya."""
+
+import logging
+
+from tuya_iot import ProjectType, TuyaOpenAPI
+import voluptuous as vol
+
+from homeassistant import config_entries
+
+from .const import (
+    CONF_ACCESS_ID,
+    CONF_ACCESS_SECRET,
+    CONF_APP_TYPE,
+    CONF_COUNTRY_CODE,
+    CONF_ENDPOINT,
+    CONF_PASSWORD,
+    CONF_PROJECT_TYPE,
+    CONF_USERNAME,
+    DOMAIN,
+    TUYA_APP_TYPE,
+    TUYA_ENDPOINT,
+    TUYA_PROJECT_TYPE,
+)
+
+RESULT_SINGLE_INSTANCE = "single_instance_allowed"
+RESULT_AUTH_FAILED = "invalid_auth"
+TUYA_ENDPOINT_BASE = "https://openapi.tuyacn.com"
+
+_LOGGER = logging.getLogger(__name__)
+
+# Project Type
+DATA_SCHEMA_PROJECT_TYPE = vol.Schema(
+    {vol.Required(CONF_PROJECT_TYPE, default=0): vol.In(TUYA_PROJECT_TYPE)}
+)
+
+# INDUSTY_SOLUTIONS Schema
+DATA_SCHEMA_INDUSTRY_SOLUTIONS = vol.Schema(
+    {
+        vol.Required(CONF_ENDPOINT): vol.In(TUYA_ENDPOINT),
+        vol.Required(CONF_ACCESS_ID): str,
+        vol.Required(CONF_ACCESS_SECRET): str,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+# SMART_HOME Schema
+DATA_SCHEMA_SMART_HOME = vol.Schema(
+    {
+        vol.Required(CONF_ACCESS_ID): str,
+        vol.Required(CONF_ACCESS_SECRET): str,
+        vol.Required(CONF_APP_TYPE): vol.In(TUYA_APP_TYPE),
+        vol.Required(CONF_COUNTRY_CODE): str,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class TuyaConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Tuya Config Flow."""
+
+    def __init__(self) -> None:
+        """Init tuya config flow."""
+        super().__init__()
+        self.conf_project_type = None
+
+    @staticmethod
+    def _try_login(user_input):
+        project_type = ProjectType(user_input[CONF_PROJECT_TYPE])
+        api = TuyaOpenAPI(
+            user_input[CONF_ENDPOINT]
+            if project_type == ProjectType.INDUSTY_SOLUTIONS
```

Comment:
Are you planning to bump the library version where the typo is addressed in a separate new PR after this PR is merged?

Reply:
Yes, I have recorded this